mautic/mautic
# Modernizr update

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Library update]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [y]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

We were using a Modernizr JS library v2.8.3 that was not published on NPM. Meaning we wouldn't be getting security vulnerability warnings if there would be some.

I'm updating the library to latest v 3.12.0 and loading it from NPM in this PR.

I was also going through all the modules we had in our custom Modernizr build and found only 2 that we actually use.

All the Modernizr checks we were loading:
```
fontface-backgroundsize-borderimage-borderradius-boxshadow-flexbox-flexboxlegacy-hsla-multiplebgs-opacity-rgba-textshadow-cssanimations-csscolumns-generatedcontent-cssgradients-cssreflections-csstransforms-csstransforms3d-csstransitions-applicationcache-canvas-canvastext-draganddrop-hashchange-history-audio-video-indexeddb-input-inputtypes-localstorage-postmessage-sessionstorage-websockets-websqldatabase-webworkers-geolocation-inlinesvg-smil-svg-svgclippaths-touch-webgl-shiv-mq-cssclasses-teststyles-testprop-testallprops-hasevent-prefixes-domprefixes
```
All the Modernizr checks we are using:
```
touch-csstransforms3d
```
Modernizr was a dependency of the [Adminre UI](http://pontualbahia.com.br/tema/gh_docs/index.html) that we are using as Mautic administration UI.

The `touch`/`no-touch` classes were used but the check was renamed to `touchevents`/`no-touchevents` in v 3. See https://modernizr.com/news/modernizr-3-new-release-site. So I updated usages for those too.

Modernizr must be custom-build so I added a `post-install` command to `package.json` that will build our custom Modernizr build according to the `modernizr-config.json` config file. There is no extra step that needs to be taken. The build is done automatically during the `npm install` command and that is being called automatically during the `composer install` command.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Run `composer install`
3. Run `bin/console mautic:assets:generate`
4. View the source code of any Mautic administration page. Notice that the `<html>` tag has these classes:
```html
<html class="csstransforms3d touchevents">
```
It can have more classes but these are the ones that we need.
